src/storage/mod.rs

```diff
@@ -503,75 +738,96 @@ where
     }
 }
 
+/// DerefMut without autoderefing.
+///
+/// Allows forcing mutable access to be explicit. Useful to implement a flagged
+/// storage where it is easier to discover sites where components are marked as
+/// mutated. Of course, individual storages can use an associated `AccessMut`
+/// type that also implements `DerefMut`, but this provides the common
+/// denominator.
+pub trait AccessMut: core::ops::Deref {
+    /// This may generate a mutation event for certain flagged storages.
+    fn access_mut(&mut self) -> &mut Self::Target;
+}
+
+impl<T: ?Sized> AccessMut for T
+where
+    T: core::ops::DerefMut,
+{
+    fn access_mut(&mut self) -> &mut Self::Target {
+        &mut *self
+    }
+}
+
 /// Used by the framework to quickly join components.
 pub trait UnprotectedStorage<T>: TryDefault {
     /// The wrapper through with mutable access of a component is performed.
-    #[cfg(feature = "nightly")]
-    type AccessMut<'a>: DerefMut<Target = T>
+    type AccessMut<'a>: AccessMut<Target = T>
```

Comment:
I switched this from `DerefMut` to a new `AccessMut` trait to allow types that are automatically deferenceable. This is for a flagged storage variant that I'm planning that would be more explicit about when a component is marked as mutated. Most cases aren't effected by this unless they are generic over the component type. `AccessMut` also still has `Deref` as a super trait like `DerefMut` does.